spec/lib/s3_storage_spec.rb

```diff
@@ -315,4 +315,29 @@
       end
     end
   end
+
+  describe "#healthy?" do
+    let(:head_bucket_params) { { bucket: bucket_name } }
+
+    it "returns true when s3 bucket is reachable" do
+      allow(s3_client).to receive(:head_bucket)
+        .with(head_bucket_params).and_return(Aws::EmptyStructure.new)
+
+      expect(storage.healthy?).to eq true
+    end
+
+    it "returns false when s3 bucket is not found" do
```

Comment:
```suggestion
    it "returns false when an AWS exception is raised" do
```

I think we should just have one test for AWS exceptions since we only catch one - it's ok if it's a specific one (like not found though).

If you take the suggestion above about using response.successful? we'd want a test for that - it looks a bit of a pain to create a Seahorse::Client::Response so may just want to set up a mock double.